src/mimir_coordinator.py

```diff
@@ -55,14 +63,21 @@ def __init__(self, relations: List[Relation]):
     def is_coherent(self):
         """Return True if the roles list makes up a coherent mimir deployment."""
         roles = self.roles()
-        return roles and set(roles).issubset(MINIMAL_DEPLOYMENT)
+        return set(roles).issuperset(MINIMAL_DEPLOYMENT)
 
     def is_recommended(self) -> bool:
-        """Return True if is a subset of the minimal deployment."""
+        """Return True if is a superset of the minimal deployment.
+
+        I.E. If all required roles are assigned, and each role has the recommended amount of units.
+        """
         roles = self.roles()
-        return bool(roles) and set(roles).issubset(MINIMAL_DEPLOYMENT)
+        # python>=3.11 would support roles >= RECOMMENDED_DEPLOYMENT
+        for role, min_n in RECOMMENDED_DEPLOYMENT.items():
+            if roles.get(role, 0) < min_n:
+                return False
+        return True
```

Comment:
Suggestion: Can we not use issuperset here?

Reply:
is_consistent does use supersets because there all we need to check is that all required roles are assigned (regardless of how many units are assigned to each role)